### ISSUE TITLE
Problem with Conditional execution of Maxima chunks in RMarkdown

### ISSUE THREAD

**Issue by alexposekany** [rcst/rim#27]
I have a file which is meant to dynamically create exercises. 
In this file I use R for statistical calculations and random number generation and Maxima as CAS through the R library ```rim```, using the following basic setup. 
```
knitr::opts_chunk$set(echo = TRUE)
options(digits=2)
library(xtable)
library(rim)
library(reticulate)  ## better Python in R
maxima.options(engine.format = "latex", 
           engine.label = TRUE,
           inline.format = "latex", 
           inline.label = FALSE)
```

I then want to switch between different functions for an exercise by randomly generating a switch through 

```{r echo=FALSE}
switch1<-FALSE;switch2<-FALSE
exercise<-sample(1:2,1)
switch(exercise,
       switch1<-TRUE,
       switch2<-TRUE)

print(c(switch1,switch2)) # check that only one is TRUE and the other FALSE
```

Then I hand this logical switch to the eval option of the respective Maxima chunk. 

```
```{maxima eval=switch1,echo=FALSE}
f:1/(x-4)+1/(y+2)+4^2*x+4*y-3;
```

```
```{maxima eval=switch2,echo=FALSE}
f:3/(2*x+1)-1/y+6*x-4*y+2;
```

However, both statements are evaluated, as if I did not set a switch, even though one is FALSE. When I use a chunk with R or python as the programming language, the switching works - only the maxima chunk does not work. I therefore conclude that it is an issue specific for rim/maxima chunks. If you could help with the issue, I would be extremely grateful.

**Comment by rcst** [comment]
Hi @alexposekany, thanks for your patience.

Chunk options `eval` (and also `echo`) are not implemented within the maxima knitr engine and are basically ignored, as you found out yourself.

If I understand correctly, you want the maxima input code to be hidden (`echo = FALSE`) and want to control whether to evaluate (and hence print) results using these chunk options. 

As for `eval`, Implementation would be easy, however implying that the unevaluated input expression doesn't receive an input label in (case the user sets `echo = TRUE`), as these are determined only after evaluation within Maxima. 

I wonder whether you are actually interested in evaluating the expressions, but hiding the results (as usually achieved by R chunk option `results = 'hide'`).  

Thanks for bringing this up, I guess most users would expect these options to work. I'll see if I can implemented those within this week. At least the `results` and `echo` options.

**Comment by rcst** [comment]
fixed in 7727f4925730f86450c1ed0f1d0bb992a9b5cdd3 and ca145eb516e48da34afcdac24740241eb61be632

**Comment by rcst** [comment]
I'll push the changes to CRAN in a while, after I manage to cover chunk option `results`. Could you pull the latest version via `devtools::install_github("rcst/rim")` and let me know if it works for you now?
